src/Symfony/Component/Security/Http/Tests/EventListener/SessionLogoutListenerTest.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Tests\EventListener;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\Security\Http\Event\LogoutEvent;
+use Symfony\Component\Security\Http\EventListener\SessionLogoutListener;
+
+class SessionLogoutListenerTest extends TestCase
+{
+    public function testOnLogoutIfHasNotSession()
+    {
+        $session = $this->createMock(Session::class);
+        $session->expects($this->never())->method('invalidate');
+
+        $request = $this->createMock(Request::class);
+        $request->expects($this->once())->method('hasSession')->willReturn(false);
```

Comment:
```suggestion
        $request->method('hasSession')->willReturn(false);
        $request->expects($this->never())->method('getSession');
```

I don't think it's important how often `hasSession()` is called, but let's make sure `getSession()` is not called.

Reply:
Should be `$request->expects($this->any())->method('hasSession')->willReturn(false);` in that case, right?

Reply:
@wouterj That's the same, just with more characters. 😉